### PR TITLE
VM-1077: Display vendor rating

### PR DESCRIPTION
**Pull request by asvishnyakov**
## Description
Depends on
* https://github.com/VirtoCommerce/vc-module-experience-api/pull/356

### Jira-link:

https://virtocommerce.atlassian.net/browse/VM-1077

### CODE REVIEW

File: src/VirtoCommerce.CustomerReviews.Web/Module.cs [hunk 1]
```diff
@@ -62,6 +68,19 @@ public void Initialize(IServiceCollection serviceCollection)
 
             serviceCollection.AddTransient<ReviewStatusChangedEventHandler>();
             serviceCollection.AddTransient<OrderChangedEventHandler>();
+
+            // GraphQL
+            var assemblyMarker = typeof(AssemblyMarker);
```
Comment (tatarincev) [inline src/VirtoCommerce.CustomerReviews.Web/Module.cs:73]: Please move all these initializations to extensions method in  the VirtoCommerce.CustomerReviews.ExperienceApi.csproj

**Review by tatarincev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CustomerReviews.Web/Module.cs (hunk 1 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerReviews.Web/Module.cs]: Done

**Review by ksavosteev — DISMISSED** [review]

File: src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        if (parameter == null)
+        {
+            throw new ArgumentNullException(nameof(parameter));
+        }
+
+        var query = parameter.Query;
+        if (query == null)
+        {
+            throw new OperationCanceledException("Query must be set");
```
Comment (artem-dudarev) [inline src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs:36]: ArgumentException

File: src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        if (responseGroup.HasFlag(ExpProductResponseGroup.LoadRating) && parameter.Results.Any())
+        {
+            var vendors = parameter.Results.Where(product => product.Vendor != null).Select(product => product.Vendor).ToArray();
+
+            var ratings = new List<RatingEntityDto>();
+
+            foreach (var vendorsByType in vendors.GroupBy(vendor => vendor.Type))
+            {
+                var vendorType = vendorsByType.Key;
+                var vendorIds = vendorsByType.Select(vendor => vendor.Id).ToArray();
```
Comment (artem-dudarev) [inline src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs:49]: Add Distinct()

File: src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                ratings .AddRange(await _ratingService.GetForStoreAsync(query.StoreId, vendorIds, vendorType));
+
```
Comment (artem-dudarev) [inline src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs:51]: Remove whitespace before `.`
Remove empty line

Enable formatting on Save

**Review by artem-dudarev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs (hunk 2 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs:36]: I follow types used in other middlewares

File: src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs (hunk 3 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs]: Fixed

File: src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs (hunk 4 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs]: Fixed. Ok :)

**Pull request merged**